File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int count =0 ;
        unordered_map<int,int> freq;
        freq[0]=1;
        int prefix = 0;
        for(int x : nums){
            prefix += x;
            int need = prefix - k;
            if(freq.count(need)){
                count += freq[need];
            }
            freq[prefix]++;
        }
        return count;
    }
};

File: strive/560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        mp = {0:1}
        sum = 0
        ans = 0
        for num in nums:
            sum += num
            need  = sum - k
            if need in mp:
                ans += mp[need]
            mp[sum] = mp.get(sum,0)+1
        return ans